core/lib/Cardano/Address/Script.hs

```diff
@@ -120,6 +127,10 @@ serializeScript script =
             , CBOR.encodeInt (fromInteger $ toInteger m)
             , encodeFoldable toCBOR contents
             ]
+        ValidFromSlot slotNum ->
+            encodeMultiscriptCtr 4 2 <> CBOR.encodeWord64 (fromInteger $ toInteger slotNum)
```

Comment:
```suggestion
            encodeMultiscriptCtr 4 2 <> CBOR.encodeWord64 (fromIntegral slotNum)
```

Reply:
we are having `Natural` as a slot (to be least surprise as we use it also for `ChainPointer`). and we need to convert to Integer and then to Word64 like that to my best knowledge